syntax = "proto3";

message PersonMessage {
  int64 id = 1;
  string first_name = 2;
  string last_name = 3;
  string company_name = 4;
}

message Empty {
}

message PersonMessageList {
  repeated PersonMessage persons = 1;
}

message RetrievePersonRequest {
  int64 id = 1;
}

service PersonService {
  rpc Create(PersonMessage) returns (PersonMessage);
  rpc Retrieve(RetrievePersonRequest) returns (PersonMessage);
  rpc RetrieveAll(Empty) returns (PersonMessageList);
}